src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -1,227 +1,280 @@
-        for (int i = 0; i < imps.size(); i++) {
-            final Imp currentImp = imps.get(i);
-            final ExtImpConsumable extImpConsumable = parseImpExt(currentImp);
-            if (i == 0) {
-                requestBuilder
-                        .networkId(extImpConsumable.getNetworkId())
-                        .siteId(extImpConsumable.getSiteId())
-                        .unitId(extImpConsumable.getUnitId())
-                        .unitName(extImpConsumable.getUnitName());
-            }
-            placements.add(ConsumablePlacement.builder()
-                    .divName(currentImp.getId())
-                    .networkId(extImpConsumable.getNetworkId())
-                    .siteId(extImpConsumable.getSiteId())
-                    .unitId(extImpConsumable.getUnitId())
-                    .unitName(extImpConsumable.getUnitName())
-                    .adTypes(ConsumableAdType.getSizeCodes(currentImp.getBanner().getFormat()))
-                    .build());
+    @Override
+    public CompositeBidderResponse makeBidderResponse(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            final List<BidderError> bidderErrors = new ArrayList<>();
+            return CompositeBidderResponse.builder().bids(extractConsumableBids(bidRequest, bidResponse, bidderErrors))
+                    .errors(bidderErrors).build();
+        } catch (DecodeException e) {
+            return CompositeBidderResponse.withError(BidderError.badServerResponse(e.getMessage()));
         }
-        requestBuilder.placements(placements);
     }
 
     private ExtImpConsumable parseImpExt(Imp imp) {
         try {
-            return mapper.mapper().convertValue(imp.getExt().get("bidder"), ExtImpConsumable.class);
+            return mapper.mapper().convertValue(imp.getExt(), CONS_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
-            throw new PreBidException(e.getMessage(), e);
+            throw new PreBidException(e.getMessage());
         }
     }
 
-    private static MultiMap resolveHeaders(BidRequest request) {
-        final MultiMap headers = HttpUtil.headers();
+    private Imp modifyImp(Imp imp, ExtImpConsumable impExt) {
+        final UpdateResult<ObjectNode> impExtUpdateResult = modifyImpExt(imp, impExt);
 
-        final Device device = request.getDevice();
-        if (device != null) {
-            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.USER_AGENT_HEADER, device.getUa());
-            final String ip = device.getIp();
-            if (StringUtils.isNotBlank(ip)) {
-                headers.add(HttpUtil.X_FORWARDED_FOR_HEADER, ip);
-                headers.add("Forwarded", "for=" + ip);
-            }
+        return imp;
+    }
+
+    private UpdateResult<ObjectNode> modifyImpExt(Imp imp, ExtImpConsumable extImpConsumable) {
```

Comment:
please order the methods according the code style: the public one first, then all the private ones that are called from the public in the order of calling, then another public one, and then its private ones and so on